### PR TITLE
feat: change from webpack to vite

### PR DESCRIPTION
**Pull request by joneugster**

### CODE REVIEW

**Comment by abentkamp** [comment]
Oh, I forgot about the Logo

**Comment by joneugster** [comment]
- [ ] Currently it's starting on `http://localhost:5173`. Can we change that back to `3000` or update the documentation?

File: server/LeanProject/.gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /build
 /lake-packages/*
+*.olean
```
Comment (joneugster) [inline server/LeanProject/.gitignore:3]:
```suggestion
lakefile.olean
```

File: package.json [hunk 2]
```diff
@@ -31,40 +32,30 @@
...
+    "vite": "^4.5.0",
     "vscode-ws-jsonrpc": "^2.0.0",
     "ws": "^8.9.0"
   },
   "devDependencies": {
-    "@babel/cli": "^7.1.0",
-    "@babel/core": "^7.1.0",
-    "@babel/preset-env": "^7.1.0",
-    "@babel/preset-react": "^7.0.0",
-    "@babel/preset-typescript": "^7.18.6",
```
Comment (joneugster) [inline package.json:42]: was babel just needed for webpack?

File: index.html
Comment (joneugster) [inline index.html:1]: I thought files from `client/public` are copied to the server. Why do we need this copy?

File: client/src/editor/leanclient.ts [hunk 3]
```diff
@@ -606,16 +606,16 @@ export class LeanClient implements Disposable {
...
+  //   const prefix = 'Lake version'
+  //   if (v.startsWith(prefix)) v = v.slice(prefix.length).trim()
+  //   const pos = v.indexOf('(')
+  //   if (pos > 0) v = v.slice(0, pos).trim()
+  //   try {
+  //     return new SemVer(v)
+  //   } catch {
+  //     return new SemVer('0.0.0')
+  //   }
+  // }
```
Comment (joneugster) [inline client/src/editor/leanclient.ts:620]: Was that a way to get the lean version? We'd need such a feature now that I disabled to `copy_versions.sh`

Comment (abentkamp) [reply client/src/editor/leanclient.ts:620]: I think that function was not used anywhere. At least the IDE showed it in gray and it's private.

File: package.json (hunk 2 above)
Comment (abentkamp) [reply package.json:42]: vite normally uses swi instead. I am not sure whether I set it up correctly, but it seems to work.

**Comment by abentkamp** [comment]
The module `path` is only meant to be used on the server side (nodejs). `webpack` replaces it automatically with `path-browserify`. I used this vite configuration to fix the issue:
https://dev.to/0xbf/vite-module-path-has-been-externalized-for-browser-compatibility-2bo6

**Comment by joneugster** [comment]
Now I get an error about the logo again.

Screenshot 2023-10-25 at 09 47 28

**Pull request merged**